daemon/api_prompting.go

```diff
@@ -97,6 +99,92 @@ func getUserID(r *http.Request) (uint32, Response) {
 	return uint32(userIDInt), nil
 }
 
+func promptingNotRunningError() *apiError {
+	return &apiError{
+		Status:  500, // Internal error
+		Message: "AppArmor Prompting is not running",
+		Kind:    client.ErrorKindAppArmorPromptingNotRunning,
+	}
+}
+
+func promptingError(err error) *apiError {
+	apiErr := &apiError{
+		Message: err.Error(),
+	}
+	switch {
+	case errorIsOneOf(err, prompting.ErrPromptsClosed, prompting.ErrRulesClosed):
+		apiErr.Status = 500
+		apiErr.Kind = client.ErrorKindInterfacesRequestsManagerClosed
```

Comment:
is this useful to the client?

Reply:
Hmm, perhaps not, if this should only happen when snapd is shutting down and is past the point of being able to respond to API requests. We really care about telling the client when prompting is not running, but that's done with its own dedicated error kind. This can probably be a generic internal error? Unless this can happen when snapd is about to shut down.

Reply:
snapd daemon stops accepting connections and requests before it stops things